### PR TITLE
cmd,config,dot,rpc: Decouple configs

### PR DESCRIPTION
**Pull request by ansermino**
This PR introduces separate config structs for the P2P and RPC services for explicit use in CLI/config file setup.

Prior to these changes we relied on marshalling the toml config file directly to the P2P/RPC configs. This limits the possible functionality of our CLI/Config packages as it forces all exported fields to be included in the TOML file. This introduces further issues when we try to have shared/global config options as it breaks `dumpConfig`.

This unblock #349.

Note: I left the DB config unchanged as it will be removed in #349.

## Changes
- Adds `RpcCfg` and `P2pCfg` types specifically for CLI/config usage
- Removes `rpc.Config` as we can just pass the params to the constructor
- Updates tests
- Regenerates example config

## Tests:
```
go test -v ./...
```

### Issues:

### CODE REVIEW

File: cmd/gossamer/configcmd.go [hunk 1]
```diff
@@ -185,17 +185,27 @@ func setP2pConfig(ctx *cli.Context, fig p2p.Config) p2p.Config {
 }
 
 // createP2PService starts a p2p network layer from provided config
```
Comment (dutterbutter) [inline cmd/gossamer/configcmd.go]: Can you update this comment to reflect changes

File: config.toml [hunk 2]
```diff
@@ -1,19 +1,13 @@
-# filename: config.toml
+[db]
+  DataDir = "/home/d/.gossamer"
```
Comment (dutterbutter) [inline config.toml]: Still unable to set DataDir from `config.toml` seems to always use default?

I don't see where any changes were made to affect that in this PR though?

Comment (ansermino) [reply config.toml]: I have fixed that in #349, please see my latest comment there

File: cmd/gossamer/configcmd_test.go [hunk 3]
```diff
@@ -180,48 +179,48 @@ func TestSetP2pConfig(t *testing.T) {
...
 			[]string{"p2pport"},
 			[]interface{}{uint(1337)},
-			p2p.Config{
+			cfg.P2pCfg{
 				BootstrapNodes: cfg.DefaultP2PBootstrap,
 				Port:           1337,
-				RandSeed:       cfg.DefaultP2PRandSeed,
-				NoBootstrap:    false,
-				NoMdns:         false,
+				//RandSeed:       cfg.DefaultP2PRandSeed,
```
Comment (noot) [inline cmd/gossamer/configcmd_test.go]: should we get rid of this commented out RandSeed?

Comment (ansermino) [reply cmd/gossamer/configcmd_test.go]: done!

**Pull request merged**